src/lib/utils.ts

```diff
@@ -99,20 +110,28 @@ export const getLevel = (points: number): number => {
   return Math.floor(points / 100) + 1;
 };
 
-// TODO: Define all ranks and logic for this
-export const getUserRank = (user: PublicProfile): string => {
-  const ranks = ['Polynomial Pita', 'Factorial Flatbread'];
-  const index = user.points % 2;
-
-  return ranks[index] ?? '';
+/**
+ * Get a user's rank based on how many points they have
+ * @param points
+ * @returns rank name
+ */
+export const getUserRank = (points: number): string => {
+  const index = Math.min(ranks.length, getLevel(points)) - 1;
+  return ranks[index] as string;
 };
```

Comment:
Is there a need to change this from taking `PublicProfile` to `points`? It seems more intuitive to `getUserRank` by passing in a user instead of a number

Reply:
hm slightly, but i'm currently getting the next rank by passing `user.points + 100`, so i think this is okay for now